### ISSUE TITLE
TextFormField and TextField's cursor cannot move at the middle of the word

### ISSUE THREAD

**Issue by siulun234** [flutter/flutter#114433]
When I am trying to insert some words in the middle of the sentence in TextFormField and TextField, it will select a range of words and I am unable to move the cursor to the correct position to insert some word in the middle. 

To fix this issue, I have to add a space at the end of the sentence and remove it immediately, then every things get work. But the issue will come again after I typed some words. 

This issue is happened in web (ios), ios and android.  

https://user-images.githubusercontent.com/4901887/199273166-69162015-fd4a-416e-b217-61a00ff2e5d4.mov

## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Type a sentence into the TextField or TextFormField
3. Try to insert some words in the middle of the sentence

**Expected results:** A cursor can move between the word, and able to insert some word into it

**Actual results:** selected a rang of word when I tap the word, and I am unable to move the cursor to insert text

<details>
<summary>Code sample</summary>

```
@override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("Flutter Demo Home Page"),
        centerTitle: true,
      ),
      body: const TextField(),
    );
  }
```

</details>

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.3.6, on macOS 12.2.1 21D62 darwin-x64, locale en-GB)
    • Flutter version 3.3.6 on channel stable at /Users/samuelma/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6928314d50 (7 days ago), 2022-10-25 16:34:41 -0400
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/samuelma/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E113
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.72.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (3 available)
    • SM N9600 (mobile) • 26dc252164217ece • android-arm64  • Android 10 (API 29)
    • macOS (desktop)   • macos            • darwin-x64     • macOS 12.2.1 21D62
      darwin-x64
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      107.0.5304.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @siulun234, I attempted to reproduce the issue but I could not. Everything worked just fine on my end. 

See the recording below.  Is there something I missed?

<details>
<summary>recording</summary>

https://user-images.githubusercontent.com/88313112/199665489-78df9dc8-b956-4a22-8412-b1720e2eae75.mp4
</details>

**Comment by siulun234** [comment]
Is it because of the version of my android device? My testing device is Android 10 (physical device).

**Comment by danagbemava-nc** [comment]
Hi @siulun234, what device are you testing this on? 

If you test this in an android emulator, does it work for you or do you still experience the bug? 

Can you also check on the latest version of `stable`/`master` to see if it makes any difference?

**Comment by siulun234** [comment]
Hi @danagbemava-nc, I am using Samsung Galaxy Note 9 (Android 10) for my testing.  

I had just test the app in the android emulator (Android 10 & 11), every things is fine. But I will get this bug when I am installed the same program in my physical device. Any things will get this happen? 

I am using the  latest version stable 3.3.6 for my testing.

**Comment by danagbemava-nc** [comment]
Hi @siulun234, you mentioned that this issue was on iOS as well. What iOS device did you test this on? 

The latest stable version is 3.3.7. You can also try on the `master` channel to see if it reproduces for you.

Thank you

**Comment by renewablezip01** [comment]
Any updates on this issue. Facing the same problems on my Samsung Galaxy S9 android 10. I'm on the stable channel version 3.3.9.

**Comment by danagbemava-nc** [comment]
Hi @renewablezip01, do you experience this issue on only Samsung devices?

**Comment by renewablezip01** [comment]
> 

I've tested on Huawei p20 and it seemed to work fine, however the  'drag pin' is buggy in that it stops after roughly 1 second of dragging.

I'm having this tap selection issue with my Samsung phone, not sure if all Samsung devices is affected by this - but there was previous commenter on this topic who had a Samsung device (Samsung Galaxy Note 9) which experienced the same issue. The TextField/TextFormField works just fine on an emulator but it is buggy on an actual device, hence why you might be having issues reproducing the bug.

I tested on other flutter production released apps on play store, and it is the exact same issue, also created a new flutter project and added just a TextField, same issue.

Furthermore, both TextField and TextFormField is affected by this.

**Comment by danagbemava-nc** [comment]
> however the 'drag pin' is buggy in that it stops after roughly 1 second of dragging.

Kindly file a new issue for this so that it can be addressed. 

> The TextField/TextFormField works just fine on an emulator but it is buggy on an actual device, hence why you might be having issues reproducing the bug.

I tested on my physical device, a Poco X3 NFC as commented in https://github.com/flutter/flutter/issues/114433#issuecomment-1301723902

> Furthermore, both TextField and TextFormField is affected by this.

TextFormField technically is only a Form wrapper around TextField.

@renewablezip01 for reference sake, can you share the flutter doctor -v output of `master` and `beta` you tried this one?

**Comment by danagbemava-nc** [comment]
@renewablezip01, thanks for the info. 

Labeling this for further investigation as I cannot reproduce this locally. 

This seems to be affecting only Samsung devices at the moment.

**Comment by renewablezip01** [comment]
Upon further inspection and looking into the https://github.com/flutter/flutter/issues/98720 issue, it appears that the phone setting "cursor control" is causing this issue. Disabling it would fix the issue but that is not a viable fix as we ought not to enforce app users to disable such setting for a fix.

**Comment by darshankawar** [comment]
> Upon further inspection and looking into the #98720 issue, it appears that the phone setting "cursor control" is causing this issue. 

This is correct. I tried this issue on S10+ device (Android 12) and was able to replicate the reported behavior, but upon checking the settings, I had cursor control enabled. Disabling it didn't replicate the reported behavior.

> Disabling it would fix the issue but that is not a viable fix as we ought not to enforce app users to disable such setting for a fix.

I am not sure if Flutter would have any control over device settings so I am not sure if there would be a fix for it in the framework.

@siulun234 Please check this and try again at your end by disabling cursor control feature (assuming you had that enabled when you found this issue).
